plugins/org.eclipse.elk.core/src/org/eclipse/elk/core/Core.melk

```diff
@@ -833,6 +833,113 @@ programmatic option scaleFactor: double {
 // requires hierarchyHandling != HierarchyHandling.INCLUDE_CHILDREN 
 }
 
+advanced option topdownLayout: boolean {
+    label "Topdown Layout"
+    description
+        "Turns topdown layout on and off. If this option is enabled, hierarchical layout will be computed first for
+         the root node and then for its children recursively. Layouts are then scaled down to fit the area provided by
+         their parents. Graphs must follow a certain structure for topdown layout to work properly. 
+         {@link TopdownNodeTypes.PARALLEL_NODE} nodes must have children of type 
+         {@link TopdownNodeTypes.HIERARCHICAL_NODE} and must define {@link topdown.hierarchicalNodeWidth} and 
+         {@link topdown.hierarchicalNodeAspectRatio} for their children. Furthermore they need to be laid out using the 
+         algorithm {@link TopdownPacking}. Hierarchical nodes require the layout option {@link nodeSize.fixedGraphSize}
+         to be true to prevent the algorithm used there from resizing the hierarchical node. This option is not 
+         supported if 'Hierarchy Handling' is set to 'INCLUDE_CHILDREN'"
+    default = false
+    targets parents
+    requires topdown.nodeType
```

Comment:
There is also a documentation field (see for example Rectpacking.melk>optimizationGoal that allows adding more information via a markdown file. I suggest doing that for topdown and adding the documentation you provided in the pull request